lib/mariaex/geometry/point.ex

```diff
@@ -0,0 +1,44 @@
+defmodule Mariaex.Geometry.Point do
+
+  @moduledoc """
+  Defines the Point struct. Implements the Ecto.Type behaviour
+  """
+
+  @type t :: %Mariaex.Geometry.Point{ coordinates: {number, number}, srid: integer }
+  defstruct coordinates: {0, 0}, srid: nil
+
+  if Code.ensure_loaded?(Ecto.Type) do
+    @behaviour Ecto.Type
+
+    def type, do: :geometry
+
+    def blank?(_), do: false
+
+    def load(%__MODULE__{} = point), do: {:ok, point}
+    def load(_), do: :error
+
+    def dump(%__MODULE__{} = point), do: {:ok, point}
+    def dump(_), do: :error
+
+    def cast(%__MODULE__{} = point), do: {:ok, point}
+    def cast(_), do: :error
+  end
+
+  @type point :: Mariaex.Geometry.Point.t
+
+  @doc """
+  Takes a Point and returns a WKB binary for MySQL
+  """
+  @spec encode(point) :: binary
+  def encode(%__MODULE__{coordinates: {x, y}, srid: srid}) do
+    srid = case srid do
+      srid when is_integer(srid) -> << srid::little-32 >>
+      _ -> << 0::little-32 >>
```

Comment:
Please can we be explicit on what is matched, if this is for the `nil` case please use `nil`. Given this would allow both `0` and `nil` to encode to `0` should `0` decode to `0` or `nil`? I am unsure which is better :smile: